src/invoices/Tags/TagsForm.js

```diff
@@ -0,0 +1,133 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import {
+  difference,
+  isEqual,
+  noop,
+  sortBy,
+} from 'lodash';
+
+import { MultiSelection } from '@folio/stripes/components';
+
+const TagsForm = ({
+  allTags,
+  entityTags,
+  onAdd,
+  onRemove,
+}) => {
+  const [savedTags, setSavedTags] = useState([]);
+
+  useEffect(() => {
+    if (!isEqual(entityTags, savedTags)) {
+      setSavedTags(entityTags);
+    }
+  }, [entityTags, savedTags]);
+
+  const onChange = useCallback(
+    (tags) => {
+      const updatedTags = tags.map(t => t.value);
+
+      if (tags.length < savedTags.length) {
+        const tag = difference(savedTags, tags.map(t => t.value));
+
+        onRemove(tag[0]);
+      } else onAdd(updatedTags);
+
+      setSavedTags(updatedTags);
+    },
+    [savedTags],
+  );
+
+  const filterItems = useCallback(
```

Comment:
it doesn't have any connections with props etc, so can be moved out of component